lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp

```diff
@@ -214,4 +225,122 @@ void LLDBMemoryReader::popLocalBuffer() {
   m_local_buffer_size = 0;
 }
 
+llvm::Optional<std::pair<uint64_t, uint64_t>>
+LLDBMemoryReader::addModuleToAddressMap(ModuleSP module) {
+  // The first available address is the mask, since subsequent images are mapped
+  // in ascending order, all of them will contain this mask.
+  uint64_t next_available_address = ARTIFICIAL_ADDRESS_MASK;
+  if (!m_range_module_map.empty()) {
+    uint64_t end_of_previous_image_address = m_range_module_map.back().first;
+    // We map the images contiguously one after the other, all with the flag bit
+    // set.
+    // The start address of the next image is the next available pointer aligned
+    // address (this works because all the end addresses are also pointer
+    // aligned).
```

Comment:
Adding +8 does not change the pointer-sized-alignment of an address. If the address was 8-byte aligned before, it's still after, and if it wasn't it won't be 8-byte-aligned afterwards either.
It's also suspicious that the resolve method then subtracts 8 again. Is this necessary at all?

Reply:
It's necessary because what we're adding 8 to is the end of the previous image. Think about it this way, the first image is mapped in the range [0x2...00 - 0x2...16], we want to map the second image the next pointer aligned address, and we know that the end of the previous image is pointer aligned, so we add 8 to it and get 0x2...24.

Similarly, when we resolve this address back to its virtual address value, we'd like to undo this operation, so we subtract the end of the previous address (0x2...16) and 8, to get back what the original virtual address is without the mask.